tmt/checks/coredump.py

```diff
@@ -0,0 +1,288 @@
+        output = invocation.guest.execute(
+            Command("cat", self.coredump_timestamp_filepath)).stdout
+
+        if not output:
+            logger.debug("Failed to read timestamp file")
+            return ResultOutcome.ERROR
+
+        start_time = output.strip()
+
+        if self._check_for_crashes(
+                invocation.guest, logger, str(invocation.check_files_path), start_time
+                ):
+            return ResultOutcome.FAIL
+
+        return ResultOutcome.PASS
+
+    def _create_coredump_timestamp(
+            self, invocation: "TestInvocation") -> bool:
+
+        self.coredump_timestamp_filepath = invocation.check_files_path / COREDUMP_TIMESTAMP_FILENAME  # noqa: E501
+        # Create timestamp directory and store current time for --since filtering
+        if not invocation.guest.execute(
+                ShellScript(f"mkdir -p {invocation.check_files_path}") and
+                ShellScript(f"date '+%Y-%m-%d %H:%M:%S' > {self.coredump_timestamp_filepath}")):
+            return False
+        return True
+
+
+@provides_check("coredump")
+class Coredump(CheckPlugin[CoredumpCheck]):
+    """
+    Check for system crashes using coredump.
+
+    The check monitors for any crashes caught by systemd-coredump during test
+    execution. This includes segmentation faults and other crashes that produce
+    core dumps. By default, any crash will cause the test to fail.
+
+    Example config with optional ignore patterns:
+
+    .. code-block:: yaml
+
+        check:
+          - how: coredump
+            ignore-patterns:
+              - 'Process.*\\(sleep\\).*dumped core'  # Ignore sleep crashes
+              - 'Package: ddcutil/2.1.2-2.fc41'      # Ignore dumps of a specific package
+
+    The patterns are matched against the full coredumpctl info output, which includes
+    fields like Process, Command Line, Signal, etc. You can use 'coredumpctl info'
+    to see the available fields and their format.
+
+    .. versionadded:: 1.41
+    """
+
+    _check_class = CoredumpCheck
+
+    @classmethod
+    def before_test(
+            cls,
+            *,
+            check: "CoredumpCheck",
+            invocation: "TestInvocation",
+            environment: Optional[tmt.utils.Environment] = None,
+            logger: tmt.log.Logger,
+            ) -> list[CheckResult]:
+        """Check for crashes before the test starts."""
+
+        if not invocation.guest.facts.has_systemd or not check._check_coredump_available(
+                invocation.guest, logger):
+            logger.debug("coredump not available, skipping..")
+            check.is_available = False
+            return [CheckResult(name="coredump", result=ResultOutcome.SKIP)]
+
+        if not check._create_coredump_timestamp(invocation):
+            outcome = ResultOutcome.ERROR
+
+        check._configure_coredump(invocation.guest, logger)
+        outcome = check._check_coredump(invocation, CheckEvent.BEFORE_TEST, logger)
```

Comment:
What is expected from checking for coredumps that appear between the timestamp and the beginning of the test?